data_steward/analytics/cdr_ops/combined.py

```diff
@@ -0,0 +1,479 @@
+           AND d.data_type = 'DATE'
+           AND ts.data_type = 'TIMESTAMP'
+       JOIN `{DATASET_ID}.INFORMATION_SCHEMA.COLUMNS` pk
+        ON ts.table_name = pk.table_name
+           AND pk.ordinal_position=1)
+
+    SELECT
+     STRING_AGG(
+      '(SELECT ' 
+         || '"' || table_name        || '" AS table_name '
+         || ',' || key_field         || '  AS row_id '
+         || ',"' || date_field       || '" AS date_field '
+         || ',' || date_field        || '  AS date_value '
+         || ',"' || timestamp_field  || '" AS timestamp_field '
+         || ',' || timestamp_field   || '  AS timestamp_value '
+         || 'FROM `' || table_schema || '.' || table_name || '` '
+         || 'WHERE ' || date_field || '<>' || 'DATE(' || timestamp_field || ') '
+         || ' LIMIT 10 )'
+     ,' UNION ALL ')
+    FROM field_comparison
+);
+EXECUTE IMMEDIATE query;
+''')
+
+q(f'''
+SELECT
+ project_id
+,dataset_id
+,table_id
+,row_count
+FROM `2021q2r1_combined_release.__TABLES__` 
+WHERE table_id = 'note'
+
+UNION ALL
+
+SELECT
+ project_id
+,dataset_id
+,table_id
+,row_count
+FROM `2021q3r1_combined_release.__TABLES__` 
+WHERE table_id = 'note'
+''')
+
+# ---
+# # Manual Review
+# The following evaluation metrics require manual/visual review.
+
+# ## Verify dataset properties
+# The dataset metadata must have
+# 1. a description indicating the version of the pipeline run and the input datasets
+# 2. labels `phase`, `release_tag`, `de_identified` with appropriate values
+#
+# In the future this should try to find the dataset using labels supplied via notebook
+# parameters (e.g. `combined`, `release_tag=2021q3r1`, `phase=clean`). Currently
+# there is no way to specify `combined`.
+
+query = f'''
+WITH schema_opts AS
+(SELECT 
+ catalog_name
+,schema_name
+,SPLIT( REPLACE( REPLACE( REPLACE( 
+        SUBSTR(option_value, 2, LENGTH(option_value)-4), 
+        "STRUCT(\\"", '' ), 
+        "\\", \\"", ":" ), 
+        "\\"), ", "," )
+      ) AS labels
+FROM `INFORMATION_SCHEMA.SCHEMATA_OPTIONS` 
+WHERE OPTION_NAME="labels")
+SELECT 
+ catalog_name
+,schema_name
+,STRUCT(
+    ARRAY_AGG(STRUCT(SPLIT(un_labels, ":")[OFFSET(0)]         AS name,
+    SPLIT(un_labels, ":")[OFFSET(1)] AS value) RESPECT NULLS) AS label_object
+ ) AS label
+FROM schema_opts
+LEFT JOIN UNNEST(labels) AS un_labels
+WHERE schema_name='{DATASET_ID}'
+GROUP BY 1, 2
+'''
+q(query)
+
+# ## Invalid concept prevalence
+# EHR submissions may provide source `concept_id`s which we assume would be
+# successfully mapped to standard `concept_id`s. This compares the frequency
+# of invalid values in required concept_id fields per hpo between the current
+# combined dataset and a baseline combined dataset. The frequency of invalid
+# values in this dataset should generally be non-increasing compared to baseline.
+# If the prevalence of invalid concepts is increasing, EHR Ops should be alerted.
+
+FQN_INVALID_CONCEPT = 'curation_sandbox.invalid_concept'
```

Comment:
I'm not sure this should be hard coded.  Why couldn't it use the combined dataset sandbox?  You could define it with `f'{DATSET_ID}_sandbox.invalid_concept`.

Reply:
Thank you! This resolved qualms I had in my use of `CREATE OR REPLACE`; with the table in the combined sandbox `CREATE IF NOT EXISTS` looks like a clearly better choice. So I made that change.